lib/go-tc/deliveryservices.go

```diff
@@ -446,8 +446,11 @@ func (ds *DeliveryServiceNullable) Validate(tx *sql.Tx) error {
 		"remapText":           validation.Validate(ds.RemapText, noLineBreaks),
 		"routingName":         validation.Validate(ds.RoutingName, isDNSName, noPeriods, validation.Length(1, 48)),
 		"typeId":              validation.Validate(ds.TypeID, validation.Required, validation.Min(1)),
-		"xmlId":               validation.Validate(ds.XMLID, noSpaces, noPeriods, validation.Length(1, 48)),
+		"xmlId":               validation.Validate(ds.XMLID, validation.NotNil, noSpaces, noPeriods, validation.Length(1, 48)),
```

Comment:
I think this might need to be `validation.Required`, which covers both the `nil` and `""` cases. So in theory the `if` below is unnecessary.